pele_platform/Frag/simulation.py

```diff
@@ -308,7 +288,91 @@ def _analysis(self):
+                self.parameters.control_file
+            )
+            if self.parameters.frag_library:
+                for path in self.parameters.working_dir:
+                    simulation_output = os.path.join(path, 'sampling_result')
+                    run_analysis(path, simulation_output)
+            else:
+                simulation_output = os.path.join(path, 'sampling_result')
+                run_analysis(self.parameters.working_dir, simulation_output)
+            return 0
+
+        if self.parameters.frag_library:
+            for sim_directory in self.parameters.working_dir:
+                simulation_output = os.path.join(self.parameters.working_dir, 'sampling_result')
+                if not os.path.isdir(simulation_output):
+                    continue
+                run_analysis(sim_directory, simulation_output)
+        else:
+            simulation_output = os.path.join(self.parameters.working_dir, 'sampling_result')
+            if not os.path.isdir(simulation_output):
+                return 0
+            run_analysis(self.parameters.working_dir, simulation_output)
+
+
+        self.parameters.analysis_to_point = \
+            self.parameters.args.analysis_to_point
+
+        if self.parameters.analysis_to_point and self.parameters.folder:
+            ana.main(path=self.parameters.folder,
+                     atom_coords=self.parameters.analysis_to_point,
+                     pattern=os.path.basename(self.parameters.system))
+
     def _clean_up(self, fragment_files):
         for file in fragment_files:
             if os.path.isfile(file):
                 os.remove(file)
+
+    def _extract_working_directory(self):
+        params = self.parameters
+        if params.frag_library:
+            params.working_dir = []
+            pdb_basename = params.core.split(".pdb")[0] + "_processed"  # Get the name of the pdb without extension
+            if "/" in pdb_basename:
+                pdb_basename = pdb_basename.split("/")[-1]  # And if it is a path, get only the name
+            current_path = os.path.abspath(".")
+
+            with open(params.input, "r") as input_file:
+                for line in input_file.readlines():
+
+                    ID = line.split('/')[-1].split(".pdb")
+                    ID = "".join(ID[:]).replace(" ", "")
+                    params.working_dir.append(os.path.join(current_path, "{}_{}".format(pdb_basename, ID)).strip('\n'))
+        else:
+            pdb_basename = params.core.split(".pdb")[0] + "_processed"  # Get the name of the pdb without extension
+            if "/" in pdb_basename:
+                pdb_basename = pdb_basename.split("/")[-1]  # And if it is a path, get only the name
+            current_path = os.path.abspath(".")
+            # If we are using an input.conf file from a previous
+            # simulation, that is, we are defining the flag frag_input
+            if "/" in self.parameters.input:
```

Comment:
Find a better way to do it (I'm sure there is one). If not, at least write some comments, this is crazy...